cell_computing_service.py

```diff
@@ -1,86 +1,67 @@
 import numpy as np
-import grpc
-import protocol_pb2 as proto
-import protocol_pb2_grpc as grpc_proto
-import cis_config as conf
 import random
 import time
-import dna_decoding
 import os
 import math
 import uuid
-import metrics
 
-import cis_env
-import cis_cell
+import grpc
+import protocol_pb2 as proto
+import protocol_pb2_grpc as grpc_proto
+
+import cis_env as env
+from cis_cell import cells_consume_energy, cells_survive, cells_divide, random_dna
+import cis_config as conf
+import dna_decoding
+import metrics
+import cis_helper as helper
 
 
 class CellComputeServicer(grpc_proto.CellInteractionServiceServicer):
     """
+    Handles computation of cells.
     """
+
     COMPUTE_CELL_INTERACTION_HISTOGRAM = metrics.request_latency_histogram.labels("compute_cell_interactions")
 
     @COMPUTE_CELL_INTERACTION_HISTOGRAM.time()
     def ComputeCellInteractions(self, incoming_batch, context):
         """
-            Computes the interaction of the whole batch of cells.
+            Computes the interaction of one batch of cells.
         """
 
-        new_cells = []
-        id_to_cell = {}
-        id_to_cell_moved = {}
-        id_to_cell_energy_averaged = {}
-        for c in incoming_batch.cells_to_compute:
-            id_to_cell[c.id] = c
-        for c in incoming_batch.cells_in_proximity:
-            id_to_cell[c.id] = c
-
-        # Movement
-        for c in incoming_batch.cells_to_compute:
-            cis_env.move_cell_and_connected_cells(
-                c, id_to_cell, id_to_cell_moved)
-
-        # Interaction
-
-        # Get Energy
-        food_fac = conf.WANTED_CELL_AMOUNT_PER_BUCKET / len(incoming_batch.cells_to_compute)
-        for c in incoming_batch.cells_to_compute:
-            cis_env.feed(
-                c,
-                incoming_batch.time_step,
-                food_factor=food_fac
-            )
+        cell_batch = incoming_batch.cells_to_compute
 
-        # Consume Energy
-        for c in incoming_batch.cells_to_compute:
-            cis_cell.consume_energy(c)
+        id_to_cell_dict = helper.map_cells_to_dict(cell_batch)
+        id_to_cell_dict = helper.map_cells_to_dict(incoming_batch.cells_in_proximity, id_to_cell_dict)
```

Comment:
`map_cells_to_dict` both changes the dict given to it and returns it(which is fine), but here it is not obvious, that dicts given to the method are modified, which can lead to mistakes later. I would rather define the dict above these two calls and dismiss the return value of the calls completely